apps/web/components/email-list/EmailPanel.tsx

```diff
@@ -29,6 +29,15 @@ export function EmailPanel(props: {
   refetch: () => void;
 }) {
   const lastMessage = props.row.messages?.[props.row.messages.length - 1];
+  const [isRead, setIsRead] = useState(false);
+  const isUnread = useMemo(() => {
+    if (lastMessage?.labelIds?.includes("UNREAD")) {
+      setIsRead(false);
+    } else {
+      setIsRead(true);
+    }
+    return lastMessage?.labelIds?.includes("UNREAD");
+  }, [lastMessage?.labelIds]);
```

Comment:
Don't think we need any extra state here. We were already marking rows as gray/white. See how that was done without state. We just get the row data passed in as props.